cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/LoadBalancingUtils.java

```diff
@@ -0,0 +1,84 @@
+import static io.cloudslang.content.amazon.entities.constants.Constants.Values.START_INDEX;
+
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.KEY;
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.MEMBER;
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.SECURITY_GROUPS;
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.SUBNETS;
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.VALUE;
+
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CustomInputs.KEY_TAGS_STRING;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.IamInputs.SECURITY_GROUP_IDS_STRING;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.NetworkInputs.SUBNET_IDS_STRING;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CustomInputs.VALUE_TAGS_STRING;
+
+/**
+ * Created by TusaM
+ * 11/10/2016.
+ */
+public class LoadBalancingUtils {
+    private static final String NAME = "Name";
+    private static final String SCHEME = "Scheme";
+    private static final String TAGS = "Tags";
+
+    public Map<String, String> getCreateLoadBalancerQueryParamsMap(InputsWrapper wrapper) {
+        Map<String, String> queryParamsMap = new LinkedHashMap<>();
+        InputsUtil.setCommonQueryParamsMap(queryParamsMap, wrapper.getCommonInputs().getAction(),
+                wrapper.getCommonInputs().getVersion());
+        queryParamsMap.put(NAME, wrapper.getLoadBalancerInputs().getLoadBalancerName());
+
+        InputsUtil.setOptionalMapEntry(queryParamsMap, SCHEME, wrapper.getLoadBalancerInputs().getScheme(),
+                Scheme.INTERNAL.name().equalsIgnoreCase(wrapper.getLoadBalancerInputs().getScheme()));
+
+        setSubnetIdQueryParams(queryParamsMap, wrapper);
+        setSecurityGroupQueryParams(queryParamsMap, wrapper);
+        setKeyAndValueTag(queryParamsMap, wrapper);
+
+        return queryParamsMap;
+    }
+
+    private void setSubnetIdQueryParams(Map<String, String> queryParamsMap, InputsWrapper wrapper) {
+        String[] subnetIdsArray = InputsUtil.getArrayWithoutDuplicateEntries(wrapper.getNetworkInputs().getSubnetIdsString(),
+                SUBNET_IDS_STRING, wrapper.getCommonInputs().getDelimiter());
+        for (int index = START_INDEX; index < subnetIdsArray.length; index++) {
+            queryParamsMap.put(SUBNETS + DOT + MEMBER + DOT + String.valueOf(index + ONE), subnetIdsArray[index]);
+        }
+    }
+
+    private void setSecurityGroupQueryParams(Map<String, String> queryParamsMap, InputsWrapper wrapper) {
+        String[] securityGroupsArray = InputsUtil.getArrayWithoutDuplicateEntries(wrapper.getIamInputs().getSecurityGroupIdsString(),
+                SECURITY_GROUP_IDS_STRING, wrapper.getCommonInputs().getDelimiter());
+        if (securityGroupsArray != null && securityGroupsArray.length > 0) {
```

Comment:
maybe `org.apache.commons.lang3.ArrayUtils.isEmpty()`ifyou have the class as dependency